library/ssl_cli.c

```diff
@@ -194,11 +225,11 @@ static void ssl_write_signature_algorithms_ext( mbedtls_ssl_context *ssl,
 #endif
     MBEDTLS_SSL_END_FOR_EACH_SIG_HASH_TLS
 
-    if( end < p || (size_t)( end - p ) < sig_alg_len + 6 )
-    {
-        MBEDTLS_SSL_DEBUG_MSG( 1, ( "buffer too small" ) );
-        return;
-    }
+    /* Empty signature algorithms list, this is a configuration error. */
+    if( sig_alg_len == 0 )
+        return( MBEDTLS_ERR_SSL_BAD_CONFIG );
+
+    MBEDTLS_SSL_CHK_BUF_PTR( p, end, sig_alg_len + 6 );
```

Comment:
Any known reason why the stopped to use MBEDTLS_SSL_CHK_BUF_PTR macro ?

Reply:
Do you mean they or us? 
It's introduced in mbedtls-2.16.7, so we weren't using it before.